host/sgx/sgxload.c

```diff
@@ -134,7 +134,18 @@ static sgx_secs_t* _new_secs(
         return NULL;
 
     memset(secs, 0, sizeof(sgx_secs_t));
+#ifdef OE_WITH_EXPERIMENTAL_EEID
+    /* This is the maximum size SGX allows. When signing base images we don't
+     * know the size that the final image will have, so we chose the maximum
+     * here. If we want to be able to use any base image for EEID, this is
+     * needed. If we decide that we want EEID to work only with special base
+     * images (like zero memory images), we can make this conditional on a flag.
+     */
+    secs->size = 68719476736;
```

Comment:
Isn't the current assumption already that EEID requires 0 memory images? Or do you mean something other than zero heap/stack size? 

This is a weird pinning to a constant value because it violates the expectation of the caller that is is setting the expected enclave size, and the enclave size as recorded in `oe_enclave_t` managed by the host still expects the enclave size to be what was calculated in  `oe_sgx_build_enclave()`.

Reply:
That's right, the _base_ image has 0 memory. The _extended_ image memory size is not known the time the base image signed. Since this value effects the MRENCLAVE, we don't want it to change between base-signing and extended-creation time, so we set this to the maximum allowable size. We could add commandline flags to enable/disable this or we could recognize 0-memory settings. If you have a better suggestion or preferred solution, I'm happy to implement it.

Reply:
I understand the reason for setting the value to max here, but I don't understand why it's valuable to retain the base image size (which has >0 memory technically, it just has no allocated heap/thread pages). Overriding the `secs->size` here creates incorrect expectations here and since the base image signing codepath needs to be special cased anyway (i.e. must be a "zero memory" image) why not just measure the expected max size into the base image?

Reply:
Yeah, that's a good point, we could just save this number during preparation of the base image.